lib/cylc/cfgspec/glbl_cfg.py

```diff
@@ -17,8 +17,15 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """Allow lazy loading of `cylc.cfgspec.globalcfg`."""
 
+CONFIG_INSTANCE = [None]
 
-def glbl_cfg():
+
+def glbl_cfg(cashed=True):
     """Load and return the global configuration singleton instance."""
     from cylc.cfgspec.globalcfg import GlobalConfig
-    return GlobalConfig.get_inst()
+    if cashed:
+        if CONFIG_INSTANCE[0] is None:
+            CONFIG_INSTANCE[0] = GlobalConfig()
+        return CONFIG_INSTANCE[0]
+    else:
+        return GlobalConfig()
```

Comment:
Can we just call `.load` on the default instance?

Reply:
We don't want to change the config that the suite was run with, Cylc does not yet posses the ability to adapt to a dynamic config? So the safer option is to return a new config object rather than updating the old one.